core/circuitbreaker/circuit_breaker.go

```diff
@@ -194,6 +210,8 @@ func (b *circuitBreakerBase) fromHalfOpenToOpen(snapshot interface{}) bool {
 		for _, listener := range stateChangeListeners {
 			listener.OnTransformToOpen(HalfOpen, *b.rule, snapshot)
 		}
+
+		stateChangeCounter.Add(float64(1), b.BoundRule().Resource, "HalfOpen", "Open")
```

Comment:
Maybe we should call stateChangeCounter.Add() in fromHalfOpenToClosed method.

Reply:
Good catch. Addressed.